src/platform/linux/misc.cpp

```diff
@@ -100,22 +100,43 @@ namespace platf {
 
   fs::path
   appdata() {
+    bool found = false;
+    bool migrate_config = true;
     const char *dir;
+    const char *homedir;
+    fs::path config_path;
+
+    // Get the home directory
+    if ((homedir = getenv("HOME")) == nullptr) {
+      homedir = getpwuid(geteuid())->pw_dir;
+    }
 
     // May be set if running under a systemd service with the ConfigurationDirectory= option set.
     if ((dir = getenv("CONFIGURATION_DIRECTORY")) != nullptr) {
-      return fs::path { dir } / "sunshine"sv;
+      found = true;
+      config_path = fs::path(dir) / "sunshine"sv;
     }
     // Otherwise, follow the XDG base directory specification:
     // https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
-    if ((dir = getenv("XDG_CONFIG_HOME")) != nullptr) {
-      return fs::path { dir } / "sunshine"sv;
+    if (!found && (dir = getenv("XDG_CONFIG_HOME")) != nullptr) {
+      found = true;
+      config_path = fs::path(dir) / "sunshine"sv;
+    }
+    // As a last resort, use the home directory
+    if (!found) {
+      migrate_config = false;
+      config_path = fs::path(homedir) / ".config/sunshine"sv;
     }
-    if ((dir = getenv("HOME")) == nullptr) {
-      dir = getpwuid(geteuid())->pw_dir;
+
+    // migrate from the old config location if necessary
```

Comment:
I'm a little worried about unconditional migration for a problem that only really affects Flatpak. I'd feel better if we gated this on an environment variable that we could set in our Flatpak.

Reply:
I'm made this block of code only run for flatpak builds using a compile definition. I think using env variables would require people changing their commands for the flatpak launch to something like this.

`flatpak override --env=VARIABLE_NAME=VARIABLE_VALUE full.application.Name`

Reply:
You can set environment variables in the Flatpak manifest like this: https://github.com/flathub/com.moonlight_stream.Moonlight/blob/master/com.moonlight_stream.Moonlight.json#L16-L21

Your approach is fine too.

Reply:
I think I like the env approach better. I will adjust it real quick.

Reply:
Updated